js/noms/src/struct.js

```diff
@@ -184,12 +191,42 @@ export class StructMirror<T: Struct> {
     newValues[i] = value;
     return newStructWithType(this.type, newValues);
   }
+
+  /**
+   * Returns a new struct where the field `name` has been removed.
+   * If `name` is not an existing field in the struct then the current struct is returned.
+   */
+  delete(name: string): Struct {
+    const oldFields = this.desc.fields;
+    const idx = findFieldIndex(name, oldFields);
+    if (idx === -1) {
+      return this._s;
+    }
+
+    // New/change field
+    const values = this._values.concat();  // clone
+    values.splice(idx, 1);
+    const type = removeFieldFromType(this.type, idx);
+    return newStructWithValues(type, values);
+  }
+}
+
+function removeFieldFromType(type: Type<StructDesc>, idx: number): Type<StructDesc> {
+  const {desc} = type;
+  const fieldMap = Object.create(null);
+  let i = 0;
+  desc.forEachField((n, t) => {
```

Comment:
Not really. Only declare types on exported bindings and where it makes the code more readable. We hardly ever annotate local variables.